packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -889,14 +894,16 @@ class _CircularProgressIndicatorState extends State<CircularProgressIndicator>
   );
 
   late AnimationController _controller;
+  late AnimationController _fallbackController;
 
   @override
   void initState() {
     super.initState();
-    _controller = AnimationController(
+    _fallbackController = AnimationController(
       duration: const Duration(milliseconds: _kIndeterminateCircularDuration),
       vsync: this,
     );
+    _controller = widget.controller ?? _fallbackController;
```

Comment:
I think a better way is to make a 
```dart
AnimationController get _controller => widget.controller ?? _fallbackController;
```
So you can safely use `_controller` for checking and building without the concern of forgetting to update this value.